test/intercept/fetch/fetch.body.browser.test.ts

```diff
@@ -0,0 +1,43 @@
+import * as path from 'path'
+import { pageWith } from 'page-with'
+
+type RequestFunction = () => Promise<Response>
+
+declare namespace window {
+  export const requestWithEmptyBody: RequestFunction
+  export const requestWithBlob: RequestFunction
+  export const requestWithFormData: RequestFunction
+  export const requestWithArrayBuffer: RequestFunction
+  export const requestWithURLSearchParams: RequestFunction
+}
+
+function prepareRuntime() {
+  return pageWith({
+    example: path.resolve(__dirname, 'fetch.body.runtime.js'),
+  })
+}
+
+test('handles requests with no body', async () => {
+  const context = await prepareRuntime()
+  await context.page.evaluate(() => window.requestWithEmptyBody())
+})
+
+test('handles requests with a Blob body', async () => {
+  const context = await prepareRuntime()
+  await context.page.evaluate(() => window.requestWithBlob())
+})
```

Comment:
As said on discord this part is not really clear and seems that test does nothing :(. Do you think there is a way to move the code from the mock to the test file ?

Reply:
Thank you for the suggestion! Assertions have been moved to the test file.